processor/batchprocessor/batch_processor.go

```diff
@@ -129,7 +129,9 @@ func newBatchProcessor(set processor.CreateSettings, cfg *Config, batchFunc func
 		metadataLimit:    int(cfg.MetadataCardinalityLimit),
 	}
 	if len(bp.metadataKeys) == 0 {
-		s := bp.newShard(nil)
+		// Retrieve info from the context and preserve Auth
+		info := client.FromContext(ctx)
```

Comment:
The context at this level doesn't have auth data, as `newBatchProcessor` isn't called in response to incoming data, but during the startup of the collector.

Reply:
Understood! The idea is to propogate the existing `context` in all shards created.

The `newShard` function is the same regardless of whether we want to start at startup with a single shard or start multiple shards as per the metadata. I figured we just need to amend the signature of `newShard` to give it an existing (if present) AuthData.

Is my understanding fair?